src/devices/machine/wd_fdc.cpp

```diff
@@ -136,6 +136,16 @@ static const char *const states[] =
 	"WRITE_SECTOR_PRE_BYTE"
 };
 
+template <unsigned B> uint32_t wd_fdc_device_base::live_info::shift_reg_low() const
+{
+	return shift_reg & make_bitmask<uint32_t>(B);
+}
+
+uint8_t wd_fdc_device_base::live_info::shift_reg_data() const
+{
+	return bitswap<8>(shift_reg, 14, 12, 10, 8, 6, 4, 2, 0);
+}
```

Comment:
You can mark these implementations `inline` to prevent the compiler from generating externally visible out-of-line bodies.